File: archaic/src/main/scala/resonantinduction/archaic/waila/WailaCrate.java
package resonantinduction.archaic.waila;

import java.util.List;

import mcp.mobius.waila.api.IWailaConfigHandler;
import mcp.mobius.waila.api.IWailaDataAccessor;
import mcp.mobius.waila.api.IWailaDataProvider;
import net.minecraft.item.ItemStack;
import net.minecraft.tileentity.TileEntity;
import resonant.lib.utility.LanguageUtility;
import resonantinduction.archaic.crate.TileCrate;

/** Waila support for crates
 * 
 * @author Darkguardsman */
public class WailaCrate implements IWailaDataProvider
{
    @Override
    public List<String> getWailaBody(ItemStack itemStack, List<String> currenttip, IWailaDataAccessor accessor, IWailaConfigHandler config)
    {
        TileEntity tile = accessor.getTileEntity();
        if (tile instanceof TileCrate)
        {
            ItemStack stored = ((TileCrate) tile).getSampleStack();
            int cap = ((TileCrate) tile).getSlotCount() * 64;
            if (stored != null)
            {
                currenttip.add(LanguageUtility.getLocal("info.waila.crate.stack") + " " + stored.getDisplayName());
                currenttip.add(LanguageUtility.getLocal("info.waila.crate.stored") + " " + stored.stackSize + " / " + cap);
            }
            else
            {
                currenttip.add(LanguageUtility.getLocal("info.waila.crate.empty"));
            }
        }
        return currenttip;
    }

    @Override
    public List<String> getWailaHead(ItemStack itemStack, List<String> currenttip, IWailaDataAccessor accessor, IWailaConfigHandler config)
    {
        return currenttip;
    }

    @Override
    public ItemStack getWailaStack(IWailaDataAccessor accessor, IWailaConfigHandler config)
    {
        return null;
    }

    @Override
    public List<String> getWailaTail(ItemStack itemStack, List<String> currenttip, IWailaDataAccessor accessor, IWailaConfigHandler config)
    {
        return currenttip;
    }

}

File: src/main/scala/resonantinduction/archaic/waila/WailaCrate.scala
package resonantinduction.archaic.waila

import mcp.mobius.waila.api.IWailaConfigHandler
import mcp.mobius.waila.api.IWailaDataAccessor
import mcp.mobius.waila.api.IWailaDataProvider
import net.minecraft.item.ItemStack
import net.minecraft.tileentity.TileEntity
import resonant.lib.utility.LanguageUtility
import resonantinduction.archaic.crate.TileCrate
import java.util.List

/**
 * Waila support for crates
 *
 * @author Darkguardsman
 */
class WailaCrate extends IWailaDataProvider
{
    override def getWailaBody(itemStack: ItemStack, currenttip: List[String], accessor: IWailaDataAccessor, config: IWailaConfigHandler): List[String] =
    {
        val tile: TileEntity = accessor.getTileEntity
        if (tile.isInstanceOf[TileCrate])
        {
            val stored: ItemStack = (tile.asInstanceOf[TileCrate]).getSampleStack
            val cap: Int = (tile.asInstanceOf[TileCrate]).getSlotCount * 64
            if (stored != null)
            {
                currenttip.add(LanguageUtility.getLocal("info.waila.crate.stack") + " " + stored.getDisplayName)
                currenttip.add(LanguageUtility.getLocal("info.waila.crate.stored") + " " + stored.stackSize + " / " + cap)
            }
            else
            {
                currenttip.add(LanguageUtility.getLocal("info.waila.crate.empty"))
            }
        }
        return currenttip
    }

    override def getWailaHead(itemStack: ItemStack, currenttip: List[String], accessor: IWailaDataAccessor, config: IWailaConfigHandler): List[String] =
    {
        return currenttip
    }

    override def getWailaStack(accessor: IWailaDataAccessor, config: IWailaConfigHandler): ItemStack =
    {
        return null
    }

    override def getWailaTail(itemStack: ItemStack, currenttip: List[String], accessor: IWailaDataAccessor, config: IWailaConfigHandler): List[String] =
    {
        return currenttip
    }
}